ops/testing.py

```diff
@@ -2249,27 +2370,34 @@ def get_path(self, path: typing.Union[str, pathlib.PurePosixPath]) \
         current_object = self.root
         for token in tokens:
             # ASSUMPTION / TESTME: object might be file
+            if isinstance(current_object, _File):
+                raise RuntimeError('cannot expand path {!r} from {!r}: '
+                                   'not a directory'.format(token, current_object))
             if token in current_object:
                 current_object = current_object[token]
             else:
                 raise FileNotFoundError(str(current_object.path / token))
         return current_object
 
-    def delete_path(self, path: typing.Union[str, pathlib.PurePosixPath]) -> None:
+    def delete_path(self, path: '_StringOrPath') -> None:
         for mount in self._mounts.values():
             if mount.contains(path):
                 return mount.delete_path(path)
         path = pathlib.PurePosixPath(path)
         parent_dir = self.get_path(path.parent)
+        if not isinstance(parent_dir, _Directory):
+            # todo: or should we be casting instead?
+            raise RuntimeError('cannot delete {}: parent {!r}'
+                               'is not a directory'.format(path.name, parent_dir))
```

Comment:
I think that isinstance feels pretty natural here. I'm not sure that casting would be any more correct.